passive/f5_bigip_cookie_internal_ip.js

```diff
@@ -0,0 +1,66 @@
+// The scan function will be called for request/response made via ZAP, excluding some of the automated tools
+// Passive scan rules should not make any requests 
+
+// Note that new passive scripts will initially be disabled
+// Right click the script in the Scripts tree and select "enable"  
+
+// kingthorin+owaspzap@gmail.com
+// Ref: https://support.f5.com/kb/en-us/solutions/public/6000/900/sol6917.html
+
+function scan(ps, msg, src) {
+	//Setup some details we will need for alerts later if we find something
+	alertRisk = 1
+	alertConfidence = 3
+	alertTitle = 'Internal IP Exposed via F5 BigIP Presistence Cookie'
+	alertDesc = 'The F5 Big-IP Persistence cookie set for this website can be decoded to a specific internal IP and port. An attacker may leverage this information to conduct Social Engineering attacks or other exploits.'
+	alertSolution = 'Configure BIG-IP cookie encryption.'
+	alertRefs = 'https://support.f5.com/kb/en-us/solutions/public/6000/900/sol6917.html'
+	cweId = 311
+	wascId = 13
+
+	url = msg.getRequestHeader().getURI().toString();
+	//Only check when a cookie is set
+	if(msg.getResponseHeader().getHeaders("Set-Cookie")) { 

```

Comment:
I purposefully decided to only implement on set-cookie in order to keep from assessing every single request, decoding the cookie value(s) over and over again, for a single alert.